ConcurrentModificationException in OkBuck 0.39.X

Went to bump OkBuck from `0.38.0` to `0.39.3` and got the following stacktrace:
```
[2018-06-18 23:53:19] * Exception is:
[2018-06-18 23:53:19] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':module:okbuck'.
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.SkipCachedTaskExecuter.execute(SkipCachedTaskExecuter.java:105)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ResolveBuildCacheKeyExecuter.execute(ResolveBuildCacheKeyExecuter.java:66)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:97)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
[2018-06-18 23:53:19] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$ExecuteTaskAction.execute(DefaultTaskExecutionGraph.java:262)
[2018-06-18 23:53:19] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$ExecuteTaskAction.execute(DefaultTaskExecutionGraph.java:246)
[2018-06-18 23:53:19] 	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:136)
[2018-06-18 23:53:19] 	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
[2018-06-18 23:53:19] 	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.execute(DefaultTaskPlanExecutor.java:201)
[2018-06-18 23:53:19] 	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.executeWithTask(DefaultTaskPlanExecutor.java:192)
[2018-06-18 23:53:19] 	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
[2018-06-18 23:53:19] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
[2018-06-18 23:53:19] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
[2018-06-18 23:53:19] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2018-06-18 23:53:19] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2018-06-18 23:53:19] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
[2018-06-18 23:53:19] 	at java.lang.Thread.run(Thread.java:748)
[2018-06-18 23:53:19] Caused by: java.util.ConcurrentModificationException
[2018-06-18 23:53:19] 	at java.util.TreeMap$PrivateEntryIterator.nextEntry(TreeMap.java:1211)
[2018-06-18 23:53:19] 	at java.util.TreeMap$KeyIterator.next(TreeMap.java:1265)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.collections.FilteredCollection$FilteringIterator.findNext(FilteredCollection.java:108)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.collections.FilteredCollection$FilteringIterator.next(FilteredCollection.java:127)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.DefaultDomainObjectCollection$IteratorImpl.next(DefaultDomainObjectCollection.java:369)
[2018-06-18 23:53:19] 	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812)
[2018-06-18 23:53:19] 	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
[2018-06-18 23:53:19] 	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:498)
[2018-06-18 23:53:19] 	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:485)
[2018-06-18 23:53:19] 	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
[2018-06-18 23:53:19] 	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
[2018-06-18 23:53:19] 	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[2018-06-18 23:53:19] 	at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:464)
[2018-06-18 23:53:19] 	at com.uber.okbuck.core.model.android.AndroidTarget.getTestOptions(AndroidTarget.java:265)
[2018-06-18 23:53:19] 	at com.uber.okbuck.composer.android.AndroidTestRuleComposer.compose(AndroidTestRuleComposer.java:59)
[2018-06-18 23:53:19] 	at com.uber.okbuck.generator.BuckFileGenerator.createRules(BuckFileGenerator.java:171)
[2018-06-18 23:53:19] 	at com.uber.okbuck.generator.BuckFileGenerator.createRules(BuckFileGenerator.java:187)
[2018-06-18 23:53:19] 	at com.uber.okbuck.generator.BuckFileGenerator.lambda$createRules$0(BuckFileGenerator.java:82)
[2018-06-18 23:53:19] 	at java.util.HashMap.forEach(HashMap.java:1289)
[2018-06-18 23:53:19] 	at com.uber.okbuck.generator.BuckFileGenerator.createRules(BuckFileGenerator.java:67)
[2018-06-18 23:53:19] 	at com.uber.okbuck.generator.BuckFileGenerator.generate(BuckFileGenerator.java:45)
[2018-06-18 23:53:19] 	at com.uber.okbuck.OkBuckGradlePlugin.lambda$null$8(OkBuckGradlePlugin.java:207)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:794)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:761)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:97)
[2018-06-18 23:53:19] 	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
[2018-06-18 23:53:19] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
```

Appears to have been introduced in the migration from groovy to java.

--- a/buildSrc/src/main/java/com/uber/okbuck/core/model/android/AndroidTarget.java
+++ b/buildSrc/src/main/java/com/uber/okbuck/core/model/android/AndroidTarget.java
@@ -42,6 +42,7 @@ import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.apache.commons.lang3.StringUtils;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.Configuration;
@@ -250,19 +251,13 @@ public abstract class AndroidTarget extends JvmTarget {
 
   @Override
   public TestOptions getTestOptions() {
+    String testTaskName =
+        VariantType.UNIT_TEST_PREFIX
+            + StringUtils.capitalize(getName())
+            + VariantType.UNIT_TEST_SUFFIX;
+    List<Test> testTasks = ImmutableList.copyOf(getProject().getTasks().withType(Test.class));
     Optional<Test> optionalTest =
-        getProject()
-            .getTasks()
-            .withType(Test.class)
-            .stream()
-            .filter(
-                test ->
-                    test.getName()
-                        .equals(
-                            VariantType.UNIT_TEST.getPrefix()
-                                + StringUtils.capitalize(getName())
-                                + VariantType.UNIT_TEST.getSuffix()))
-            .findFirst();
+        testTasks.stream().filter(test -> test.getName().equals(testTaskName)).findFirst();
 
     List<String> jvmArgs =
         optionalTest.map(Test::getAllJvmArgs).orElseGet(Collections::<String>emptyList);
@@ -416,9 +411,9 @@ public abstract class AndroidTarget extends JvmTarget {
 
       if (manifests.size() == 1 && getMergeType() == ManifestMerger2.MergeType.LIBRARY) {
         // No need to merge for libraries
-        parseManifest(
-            Files.lines(mainManifest.toPath()).collect(Collectors.joining(System.lineSeparator())),
-            mergedManifest);
+        try (Stream<String> lines = Files.lines(mainManifest.toPath())) {
+          parseManifest(lines.collect(Collectors.joining(System.lineSeparator())), mergedManifest);
+        }
       } else {
         // always merge if more than one manifest or its an application
         List<File> secondaryManifests =

--- a/dependencies.gradle
+++ b/dependencies.gradle
@@ -1,5 +1,5 @@
 def versions = [
-        androidPlugin      : "3.1.2",
+        androidPlugin      : "3.2.0-beta01",
         androidTools       : "26.1.2",
         butterKnife        : "9.0.0-20180416.155017-37",
         butterKnifeCompiler: "9.0.0-20180416.155025-37",

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.7-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.8.1-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/transform-cli/src/main/java/com/uber/okbuck/transform/FileUtil.java
+++ b/transform-cli/src/main/java/com/uber/okbuck/transform/FileUtil.java
@@ -6,17 +6,15 @@ import java.nio.file.FileVisitOption;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Comparator;
+import java.util.stream.Stream;
 
 final class FileUtil {
 
   private FileUtil() {}
 
   static void deleteDirectory(File path) {
-    try {
-      Files.walk(path.toPath(), FileVisitOption.FOLLOW_LINKS)
-          .sorted(Comparator.reverseOrder())
-          .map(Path::toFile)
-          .forEach(File::delete);
+    try (Stream<Path> walk = Files.walk(path.toPath(), FileVisitOption.FOLLOW_LINKS)) {
+      walk.sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
     } catch (IOException ignored) {
     }
   }

